src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -696,6 +750,24 @@ public void deleteDBs() throws Exception {
     }
   }
 
+  /**
+   * This is called by operations outside of the ReaderMetricsProcessor.
+   *
+   * @return A list of the timestamps associated with MetricsDB files. The oldest of the MetricsDB files typically
+   *     have a lifetime of ~SAMPLING_INTERVAL seconds (no less than SAMPLING_INTERVAL/2 seconds). Null if batch
+   *     metrics is disabled.
+   */
+  public NavigableSet<Long> getBatchMetrics() {
+    if (batchMetricsEnabled) {
+      TreeSet<Long> batchMetricsDBSetCopy = new TreeSet<>(batchMetricsDBSet.clone());
+      while (batchMetricsDBSetCopy.size() > PluginSettings.instance().getBatchMetricsRetentionPeriod() * 12) {
```

Comment:
Add a comment about why you're using 12 b/c of the 5s snapshots. Also make it evident that the retention period is in minutes

Reply:
An understanding of the 5s snapshots should be obvious to the reader of this code. The connection to the 12 should be pretty readily understood.
Agreed on the minutes unit for the retention period -- the method itself will be renamed to say "Minutes". This should further make the connection for the first part obvious.